### PR TITLE
GATE-3151: adds support for egress policies

### PR DESCRIPTION
**Pull request by Michael9127**
## Description

Adds support for new Gateway Egress Policy rule type

## Types of changes

What sort of change does your code introduce/modify?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [X] This change is using publicly documented (api.cloudflare.com or developers.cloudflare.com) and stable APIs.

[1]: https://help.github.com/articles/closing-issues-using-keywords/

### CODE REVIEW

File: teams_rules.go [hunk 1]
```diff
@@ -35,6 +35,14 @@ type TeamsRuleSettings struct {
...
 	// whether to disable dnssec validation for allow action
 	InsecureDisableDNSSECValidation bool `json:"insecure_disable_dnssec_validation"`
+
+	EgressSettings *EgressSettings `json:"egress"`
+}
+
+type EgressSettings struct {
+	Ipv6Range    string `json:"ipv6"`
+	Ipv4         string `json:"ipv4"`
+	Ipv4Fallback string `json:"ipv4_fallback"`
```
Comment (jacobbednarz) [inline teams_rules.go:45]:
```suggestion
	IPv6         string `json:"ipv6"`
	IPv4         string `json:"ipv4"`
	IPv4Fallback string `json:"ipv4_fallback"`
```

any reason why the struct field is IPv6Range and not IPv6 given the JSON key?

**Review by jacobbednarz — COMMENTED** [review]
a nitpick about field naming. could you link me to the public docs for this one? my search fu is failing.

**Comment by Michael9127** [comment]
Api docs are now live! https://api.cloudflare.com/#zero-trust-gateway-rules-list-zero-trust-gateway-rules

File: teams_rules.go (hunk 1 above)
Comment (Michael9127) [reply teams_rules.go:45]: The ipv6 field does indeed have a cidr, the ipv4 does not. I think the json key was renamed at someones request for brevity on the wire 🤷

**Comment by jacobbednarz** [comment]
thanks 🍭

**Pull request merged**